File: ToC_2026/8/question4/Java/1.java
import java.util.Scanner;

public class Q1_Fac {
    static boolean iszero(int m) { return m == 0; }
    static int pi(int m) { return m == 0 ? 0 : m - 1; }

    static int fac(int m) {
        if (iszero(m)) {
            return 1;
        } else {
            return m * fac(pi(m));
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter a number: ");
        int m = scanner.nextInt();
        System.out.println("The factorial of " + m + " is " + fac(m));
    }
}

File: ToC_2026/8/question4/Haskell/1.hs
//階乗関数 fac(m) = m!
iszero :: Int -> Bool
iszero m = m == 0

pi_ :: Int -> Int
pi_ m = if m == 0 then 0 else m - 1

fac :: Int -> Int
fac m = if iszero m then 1 else m * fac (pi_ m)

main :: IO ()
main = do
    putStrLn "Enter a number: "
    m <- readLn
    putStrLn $ "The factorial of " ++ show m ++ " is " ++ show (fac m)